## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 526365`
Commit: [`fb25d5c02bfbd00bd6fb6d0bea2828b67025bcca`](https://github.com/NixOS/nixpkgs/commit/fb25d5c02bfbd00bd6fb6d0bea2828b67025bcca) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/fb25d5c02bfbd00bd6fb6d0bea2828b67025bcca..pull/526365/head))
Merge: [`1a4bb2a0866f653a7b6df5e43e1f8e9bf26bfc02`](https://github.com/NixOS/nixpkgs/commit/1a4bb2a0866f653a7b6df5e43e1f8e9bf26bfc02)

Logs: https://github.com/bachp/nixpkgs-review-gha/actions/runs/26722346624

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-526365-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://bachp-nixpkgs-review-gha.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  bachp-nixpkgs-review-gha.cachix.org-1:CI/v+Fc0RnhFck42q60eNjzTJ2fp6HglnI75SuE4PeY=
  ' \
  /nix/store/qp0x0mqjpwavwrwh0z2mxwiv7m65wps7-python3.13-aplpy-2.2.0 \
  /nix/store/f9c77ss8iv59zxnl6xhl1gq85q1bnyi0-python3.13-aplpy-2.2.0-dist \
  /nix/store/ab7xkgxnih83wd09hp2hp9lf6i7z4llm-python3.13-dask-image-2025.11.0 \
  /nix/store/kl902lw1dvjxfam5wlp96drabhpmj5gx-python3.13-dask-image-2025.11.0-dist \
  /nix/store/samxsjmba12vfmqzb80lp7i39zwldwv4-python3.13-multiscale-spatial-image-2.1.0 \
  /nix/store/51g2yxzn8mbqky392bi0gqhzwwgnsrir-python3.13-multiscale-spatial-image-2.1.0-dist \
  /nix/store/frmrkhc2zy9whvk0jfxij6vsw9vmzl5h-python3.13-ngff-zarr-0.34.0 \
  /nix/store/dzs0gv1d3ihmamh4dr375qd53iwddjq8-python3.13-ngff-zarr-0.34.0-dist \
  /nix/store/8y1i754sygvdlfk4783yaazw1qxr71wi-python3.13-pims-0.7 \
  /nix/store/i4x6vyn57p6ap5ccz3nqd1f4gw59891c-python3.13-pims-0.7-dist \
  /nix/store/a0cbbvdh9yk52s9awzhm33qxh0m5ds9j-python3.13-pvextractor-0.4 \
  /nix/store/mn1z9q7k4lvy2q0y6sid3c9j4mcyh281-python3.13-pvextractor-0.4-dist \
  /nix/store/cv69p2y4jlqpv19knh0skymyv6hdqcsb-python3.13-reproject-0.19.0 \
  /nix/store/4g8g3iacvh6vhp11j08mjciqa9gr3bj0-python3.13-reproject-0.19.0-dist \
  /nix/store/v1j3v74wv7nrlz34ph2i6dr65mpr6kfl-python3.13-spectral-cube-0.6.7 \
  /nix/store/6a4db0m033n6lfsgijwfipzwdf6c34m7-python3.13-spectral-cube-0.6.7-dist \
  /nix/store/53a9g4fkbcfml9ymrmd00npm5kyyfkcy-python3.13-stytra-0.8.34 \
  /nix/store/hnvykmh8600qhjvxjyhgw2f4s7f2m9pl-python3.13-stytra-0.8.34-dist \
  /nix/store/xhap0xbh7jw06bi3vvx6mgms846nlcaf-python3.14-aplpy-2.2.0 \
  /nix/store/yhhpx6hiivhma0p7r2x1vah9dv4wi5x5-python3.14-aplpy-2.2.0-dist \
  /nix/store/kqk3yr5d5rcwpp4ab0hlq9d8xz7pckfd-python3.14-dask-image-2025.11.0 \
  /nix/store/nx73ladwvq8b27iaixbmv0l4ck42dk62-python3.14-dask-image-2025.11.0-dist \
  /nix/store/jamjhphmliwvjzdwqycqk10fk289qkhw-python3.14-multiscale-spatial-image-2.1.0 \
  /nix/store/s41agqaix48msyyxyd04d3rysqgaqmj7-python3.14-multiscale-spatial-image-2.1.0-dist \
  /nix/store/3zwajfj3wh2s8ws8ygddpjx1q3p8l97j-python3.14-ngff-zarr-0.34.0 \
  /nix/store/asmgb751l647mdzaf32nrimk2k2n94qi-python3.14-ngff-zarr-0.34.0-dist \
  /nix/store/zbpfyxwl35gbiwzzzl62qw84533ksxy5-python3.14-pims-0.7 \
  /nix/store/zxmy6v4m9m9i88wckckkc1h4mmqy9gnb-python3.14-pims-0.7-dist \
  /nix/store/saby8giqqk9pgxz5595l9406mz4r6mh3-python3.14-pvextractor-0.4 \
  /nix/store/gbl5izrld4nr4hvlvdq2a0fx4wyzf5wq-python3.14-pvextractor-0.4-dist \
  /nix/store/f5bm8i94x6c8x1wbva80n6fhgcrjd10m-python3.14-reproject-0.19.0 \
  /nix/store/2c6972xqh2c229gvhc7j3iry38pgh098-python3.14-reproject-0.19.0-dist \
  /nix/store/nvnrrwcf826jjxwgrvg07mk0paq8b3q7-python3.14-spectral-cube-0.6.7 \
  /nix/store/0ni2zwrv2jzvhqnpq04bbx30waq64bb9-python3.14-spectral-cube-0.6.7-dist \
  /nix/store/pn95gqawqh7qi0bczbpmwn53p2jr3y01-python3.14-stytra-0.8.34 \
  /nix/store/wsc4f4bib7asxrxgwqlflkds4drhdmrd-python3.14-stytra-0.8.34-dist \
  /nix/store/akppasbry3cq2p93yv31fygwdzqd9pcc-python3.14-sunpy-7.1.0 \
  /nix/store/r46llrjhscz9pyfc7hxw485dz5kcvv1q-python3.14-sunpy-7.1.0-dist
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-526365-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://bachp-nixpkgs-review-gha.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  bachp-nixpkgs-review-gha.cachix.org-1:CI/v+Fc0RnhFck42q60eNjzTJ2fp6HglnI75SuE4PeY=
  ' \
  /nix/store/d4mial09cj4a5gyzkkblc4vav3cjx26k-python3.13-aplpy-2.2.0 \
  /nix/store/xndinp04hw1q5i4cgkfvrqrpdjgg4wb1-python3.13-aplpy-2.2.0-dist \
  /nix/store/6z79bvf5gcfwnkmgpq5jaj77zgj92w3b-python3.13-dask-image-2025.11.0 \
  /nix/store/aplbf35gsdlvh7iypan5mzqmgx8jvdxf-python3.13-dask-image-2025.11.0-dist \
  /nix/store/icq3nwdg8v1mmrqyrxp171ybyhmb4bdk-python3.13-pims-0.7 \
  /nix/store/brkx8b2cah5321zjr74s0255k5mk9n45-python3.13-pims-0.7-dist \
  /nix/store/w979zfha15bflzwdbrf7ngf7xyxmkpzb-python3.13-pvextractor-0.4 \
  /nix/store/cpnrzpabp5n26p52afz35ziywy3sbval-python3.13-pvextractor-0.4-dist \
  /nix/store/gxfan9plriqasffvdcfhav19mpvis172-python3.13-reproject-0.19.0 \
  /nix/store/8xs0gng9gb9g1y84w67gv0w6zw7v32si-python3.13-reproject-0.19.0-dist \
  /nix/store/fc9v69ddcy6szm8frzd1wdd5gz6p5ahx-python3.13-spectral-cube-0.6.7 \
  /nix/store/yhym053is0k59cw4xfjkiiail5k8wmc6-python3.13-spectral-cube-0.6.7-dist \
  /nix/store/dyl30ncrhvir23aqy0zrxav0295ljxwq-python3.13-stytra-0.8.34 \
  /nix/store/yi8p5djmh8j6yi2s2yar718c509p41sy-python3.13-stytra-0.8.34-dist \
  /nix/store/gmz6bzkd8jci2z8gr89jnh3nmzyjbrmg-python3.14-aplpy-2.2.0 \
  /nix/store/wh4xa3p1nqskcvw8jvsm6zy51jk50zab-python3.14-aplpy-2.2.0-dist \
  /nix/store/7j454h2r7i16r9rxvp6lf46b6lvclc8l-python3.14-dask-image-2025.11.0 \
  /nix/store/vwv7lzq1y5d9kg0x6k2ivz1ksv1hjpgl-python3.14-dask-image-2025.11.0-dist \
  /nix/store/qfgwip4bjv9n3sii2ijy9jh474xqqa3f-python3.14-pims-0.7 \
  /nix/store/vb9s19zrxmdspqr3dz5j1z0mip8222js-python3.14-pims-0.7-dist \
  /nix/store/x2wfn3k1y8kyqfcdkiw2618b0rj5pbb2-python3.14-pvextractor-0.4 \
  /nix/store/3hilg7mx7a305wxys4wq2gvp2whm27cp-python3.14-pvextractor-0.4-dist \
  /nix/store/aakjg2z4dxmxvpd8cj7hrkdh3agibhlp-python3.14-reproject-0.19.0 \
  /nix/store/63zhhh56pfpb24h9y196pwpgxqn0rwzp-python3.14-reproject-0.19.0-dist \
  /nix/store/1wxhgx82wjdwi3xzchnlzca0zadqzmd7-python3.14-spectral-cube-0.6.7 \
  /nix/store/q9nkm5x6iasji9lb9236yssimjbq9j7a-python3.14-spectral-cube-0.6.7-dist \
  /nix/store/f42rvr2r26ymqk31glwinmc55pyl1smp-python3.14-stytra-0.8.34 \
  /nix/store/rxjczcji8lflslqzz1lbkrwmbi78n65q-python3.14-stytra-0.8.34-dist \
  /nix/store/pchbq04b2rvh0q990my8mx48kk7gwzd6-python3.14-sunpy-7.1.0 \
  /nix/store/7r3mjkmvbpshnskxvz2fb74lgxjpnbbl-python3.14-sunpy-7.1.0-dist
```
</details></li>
<li><details><summary><code>aarch64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-526365-aarch64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://bachp-nixpkgs-review-gha.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  bachp-nixpkgs-review-gha.cachix.org-1:CI/v+Fc0RnhFck42q60eNjzTJ2fp6HglnI75SuE4PeY=
  ' \
  /nix/store/22ww2la3jahkcrgvqqiz081s7jlmhfds-python3.13-aplpy-2.2.0 \
  /nix/store/pf5b2f49ij4r4ggrk4dxnxnqrsc2kmvy-python3.13-aplpy-2.2.0-dist \
  /nix/store/pklwfg84hbyzwyayl91l4vncj3x1wy3k-python3.13-dask-image-2025.11.0 \
  /nix/store/ldzwg9f1z7czcyasgfh4adksa65i8bag-python3.13-dask-image-2025.11.0-dist \
  /nix/store/vs9a0gppkby4lhrgj3q3j5g5j4d5x0ii-python3.13-pims-0.7 \
  /nix/store/bfyzbmig07cy961zsh46m4ll222rr91c-python3.13-pims-0.7-dist \
  /nix/store/sbmb1ms1rivzkkh8d675w0biqgski39c-python3.13-reproject-0.19.0 \
  /nix/store/d12fpm78g48p0zy5wgwcd4ibgli8x0m9-python3.13-reproject-0.19.0-dist \
  /nix/store/1prbgp2xb2lc9ckzy50zzl8bz2fmgv5p-python3.13-spectral-cube-0.6.7 \
  /nix/store/dzb09w61rp9y8g48rbzc7qmp3640mvb3-python3.13-spectral-cube-0.6.7-dist \
  /nix/store/vr5ml0jpkr8p41jn55nblixqvv1bkqsx-python3.14-aplpy-2.2.0 \
  /nix/store/y89wpld0v1b0lx4123fggjjv00cdwvq4-python3.14-aplpy-2.2.0-dist \
  /nix/store/hkclxg6xjd1xapfpngb6kab08psixbjv-python3.14-dask-image-2025.11.0 \
  /nix/store/iplpnx1jdd5x9fgkxq6vjw5dfwvil7hp-python3.14-dask-image-2025.11.0-dist \
  /nix/store/8zw8kyrggk9f3gwixbrd2f4cqsmgy0wf-python3.14-pims-0.7 \
  /nix/store/j8w6n48y9r145klrhmny0x456jybzrkc-python3.14-pims-0.7-dist \
  /nix/store/kapaxp17d4ka0vcsd4mgq3w59xn11wyy-python3.14-reproject-0.19.0 \
  /nix/store/pdwqsbl3rmkxqkw2mcmnniqpb6q7hjhk-python3.14-reproject-0.19.0-dist \
  /nix/store/id7sdc0qihq0gwyf2kgapqywws4cswmf-python3.14-spectral-cube-0.6.7 \
  /nix/store/dfxpgq9hq7r4vd6gdfmr5bp27iw91hd5-python3.14-spectral-cube-0.6.7-dist \
  /nix/store/d8w5z8cxkcls280jmpx1lynb9q9rczdm-python3.14-sunpy-7.1.0 \
  /nix/store/62is1hw6cr8mwkz34apjdqds49bfq2cz-python3.14-sunpy-7.1.0-dist
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>python313Packages.sunpy</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 19 packages built:</summary>
  <ul>
    <li>python313Packages.aplpy</li>
    <li>python313Packages.dask-image</li>
    <li>python313Packages.multiscale-spatial-image</li>
    <li>python313Packages.ngff-zarr</li>
    <li>python313Packages.pims</li>
    <li>python313Packages.pvextractor</li>
    <li>python313Packages.reproject</li>
    <li>python313Packages.spectral-cube</li>
    <li>python313Packages.stytra</li>
    <li>python314Packages.aplpy</li>
    <li>python314Packages.dask-image</li>
    <li>python314Packages.multiscale-spatial-image</li>
    <li>python314Packages.ngff-zarr</li>
    <li>python314Packages.pims</li>
    <li>python314Packages.pvextractor</li>
    <li>python314Packages.reproject</li>
    <li>python314Packages.spectral-cube</li>
    <li>python314Packages.stytra</li>
    <li>python314Packages.sunpy</li>
  </ul>
</details>

---
<details>
<summary>python313Packages.sunpy</summary>
<pre>../../nix/store/w16y0pnha1p2bkbsksilrwrw7l7n3hnx-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/w16y0pnha1p2bkbsksilrwrw7l7n3hnx-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: asdf_schema_tests_enabled
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

../../nix/store/w16y0pnha1p2bkbsksilrwrw7l7n3hnx-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/w16y0pnha1p2bkbsksilrwrw7l7n3hnx-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: mpl-results-path
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

../../nix/store/w16y0pnha1p2bkbsksilrwrw7l7n3hnx-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/w16y0pnha1p2bkbsksilrwrw7l7n3hnx-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: mpl-use-full-test-name
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= short test summary info ==============================================================================
FAILED sunpy/util/tests/test_logger.py::test_sunpy_warnings_logging - assert 2 == 1
 +  where 2 = len(WarningsChecker(record=True))
=============================================== 1 failed, 1343 passed, 183 skipped, 113 deselected, 4 warnings in 107.19s (0:01:47) ================================================</pre>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>python313Packages.sunpy</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 15 packages built:</summary>
  <ul>
    <li>python313Packages.aplpy</li>
    <li>python313Packages.dask-image</li>
    <li>python313Packages.pims</li>
    <li>python313Packages.pvextractor</li>
    <li>python313Packages.reproject</li>
    <li>python313Packages.spectral-cube</li>
    <li>python313Packages.stytra</li>
    <li>python314Packages.aplpy</li>
    <li>python314Packages.dask-image</li>
    <li>python314Packages.pims</li>
    <li>python314Packages.pvextractor</li>
    <li>python314Packages.reproject</li>
    <li>python314Packages.spectral-cube</li>
    <li>python314Packages.stytra</li>
    <li>python314Packages.sunpy</li>
  </ul>
</details>

---
<details>
<summary>python313Packages.sunpy</summary>
<pre>../../nix/store/aqn0q8wpg6biijl07fz6321f7bdiibm1-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/aqn0q8wpg6biijl07fz6321f7bdiibm1-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: asdf_schema_tests_enabled
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

../../nix/store/aqn0q8wpg6biijl07fz6321f7bdiibm1-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/aqn0q8wpg6biijl07fz6321f7bdiibm1-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: mpl-results-path
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

../../nix/store/aqn0q8wpg6biijl07fz6321f7bdiibm1-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/aqn0q8wpg6biijl07fz6321f7bdiibm1-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: mpl-use-full-test-name
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= short test summary info ==============================================================================
FAILED sunpy/util/tests/test_logger.py::test_sunpy_warnings_logging - assert 2 == 1
 +  where 2 = len(WarningsChecker(record=True))
===================================================== 1 failed, 1343 passed, 183 skipped, 113 deselected, 4 warnings in 20.87s =====================================================</pre>
</details>

---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 5 packages failed to build:</summary>
  <ul>
    <li>python313Packages.pvextractor</li>
    <li>python313Packages.stytra</li>
    <li>python313Packages.sunpy</li>
    <li>python314Packages.pvextractor</li>
    <li>python314Packages.stytra</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>python313Packages.aplpy</li>
    <li>python313Packages.dask-image</li>
    <li>python313Packages.pims</li>
    <li>python313Packages.reproject</li>
    <li>python313Packages.spectral-cube</li>
    <li>python314Packages.aplpy</li>
    <li>python314Packages.dask-image</li>
    <li>python314Packages.pims</li>
    <li>python314Packages.reproject</li>
    <li>python314Packages.spectral-cube</li>
    <li>python314Packages.sunpy</li>
  </ul>
</details>

---
<details>
<summary>python313Packages.sunpy</summary>
<pre>../../../store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: asdf_schema_tests_enabled
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

../../../store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: mpl-results-path
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

../../../store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: mpl-use-full-test-name
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= short test summary info ==============================================================================
FAILED sunpy/util/tests/test_logger.py::test_sunpy_warnings_logging - assert 2 == 1
 +  where 2 = len(WarningsChecker(record=True))
===================================================== 1 failed, 1343 passed, 183 skipped, 113 deselected, 4 warnings in 30.23s =====================================================</pre>
</details>

--- a/pkgs/development/python-modules/pims/default.nix
+++ b/pkgs/development/python-modules/pims/default.nix
@@ -7,30 +7,54 @@
   pytestCheckHook,
   scikit-image,
   slicerator,
+  packaging,
+  tifffile,
+  jinja2,
+  jpype1,
+  matplotlib,
+  moviepy,
+  setuptools,
 }:
 
-buildPythonPackage rec {
+buildPythonPackage (finalAttrs: {
   pname = "pims";
   version = "0.7";
-  format = "setuptools";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "soft-matter";
     repo = "pims";
-    tag = "v${version}";
+    tag = "v${finalAttrs.version}";
     hash = "sha256-3SBZk11w6eTZFmETMRJaYncxY38CYne1KzoF5oRgzuY=";
   };
 
-  propagatedBuildInputs = [
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
     slicerator
     imageio
     numpy
+    packaging
+    tifffile # imported within try-excet block so optional but setup.py requires it.
   ];
+
+  optional-dependencies = {
+    # CI says its extras
+    extras = [
+      jinja2
+      jpype1
+      matplotlib
+      moviepy
+      scikit-image
+    ];
+  };
 
   nativeCheckInputs = [
     pytestCheckHook
-    scikit-image
-  ];
+  ]
+  ++ finalAttrs.passthru.optional-dependencies.extras;
 
   pythonImportsCheck = [ "pims" ];
 
@@ -46,13 +70,16 @@ buildPythonPackage rec {
   disabledTestPaths = [
     # AssertionError: Tuples differ: (377, 505, 4) != (384, 512, 4)
     "pims/tests/test_display.py"
+
+    # tests require internet connection
+    "pims/tests/test_bioformats.py"
   ];
 
   meta = {
     description = "Module to load video and sequential images in various formats";
     homepage = "https://github.com/soft-matter/pims";
-    changelog = "https://github.com/soft-matter/pims/releases/tag/v${version}";
+    changelog = "https://github.com/soft-matter/pims/releases/tag/v${finalAttrs.version}";
     license = lib.licenses.bsd3;
     maintainers = [ ];
   };
-}
+})

